integrations/pagerduty/clients/pagerduty.py

```diff
@@ -143,9 +162,11 @@ async def create_webhooks_if_not_exists(self) -> None:
         }
 
         try:
-            await self.http_client.post(
-                f"{self.api_url}/webhook_subscriptions", json=body
-            )
+            async with self._semaphore:
+                response = await self.http_client.post(
+                    f"{self.api_url}/webhook_subscriptions", json=body
+                )
+                await self._handle_rate_limiting(response)
```

Comment:
instead of having to duplicate the handling of the rate limits with the semaphores, I will recommend we have a general method like send_api_request(method="GET", params, body) 

where we handle the rate limit there.

check the SonarQube integration for sample inspiration of this method https://github.com/port-labs/ocean/blob/83d1ff14ce1c7337eb14c6231b088207e9c8d3c6/integrations/sonarqube/client.py#L92

and the Sentry for a similar rate limit implementation https://github.com/port-labs/ocean/blob/main/integrations/sentry/clients/sentry.py